Assets/LeapMotionModules/Interaction/Examples/TransformHandles/Scripts/ColorOnGrab.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Leap.Unity.UI.Interaction;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.Examples.TransformHandles {
+
+  [RequireComponent(typeof(InteractionBehaviour))]
+  public class ColorOnGrab : MonoBehaviour {
+
+    public string materialColorProperty = "_Color";
+    public Color grabbedColor = Color.white;
+    public string materialEmissionProperty = "_EmissionColor";
+    public Color grabbedEmission = Color.white;
+
+    private InteractionBehaviour _interaction;
+
+    private Material _materialInstance;
+    private Color _defaultColor;
+    private Color _defaultEmission;
+    private int _materialColorPropertyId;
+    private int _materialEmissionPropertyId;
+
+    void OnValidate() {
+      _materialColorPropertyId = Shader.PropertyToID(materialColorProperty);
```

Comment:
This OnValidate seems unnecessary since you grab the id's inside of Start anyway.